app/over_react_redux/todo_client/lib/src/components/app_bar/app_bar_local_storage_menu.dart

```diff
@@ -0,0 +1,146 @@
+  String currentDataSetName;
+  List<String> savedLocalStorageStateNames;
+  bool currentDataHasBeenModified;
+}
+
+@Component2()
+class AppBarLocalStorageMenuComponent extends UiComponent2<AppBarLocalStorageMenuProps> {
+  final _overlayRef = createRef<MenuOverlayComponent>();
+  bool get _currentStateKeyIsReadOnly => TodoAppLocalStorage.reservedStateKeys.contains(props.currentDataSetName);
+
+  @override
+  render() {
+    return (MenuOverlay()
+      ..modifyProps(addUnconsumedProps)
+      ..trigger = Button({'color': 'inherit'}, 'Data: ${props.currentDataSetName}')
+      ..useDerivedMaxWidth = true
+      ..ref = _overlayRef
+    )(
+      _renderSaveMenuItem(),
+      _renderSaveAsMenuItem(),
+      MenuItem({
+        'onClick': (_) {
+          _loadFromLocalStorage(TodoAppLocalStorage.defaultStateKey);
+        },
+      }, 'Load Default Data'),
+      MenuItem({
+        'onClick': (_) {
+          _loadFromLocalStorage(TodoAppLocalStorage.emptyStateKey);
+        },
+      }, 'Load Empty Data'),
+      _renderSavedStateMenuItems(),
+    );
+  }
+
+  ReactElement _renderSaveMenuItem() {
+    if (!props.currentDataHasBeenModified || _currentStateKeyIsReadOnly) return null;
+
+    return MenuItem({
+      'onClick': (SyntheticMouseEvent event) {
+        event.stopPropagation(); // Don't close the menu
+        _handleCurrentLocalStorageStateSaveAs(newStateName: props.currentDataSetName);
+      }
+    },
+      'Save ${props.currentDataSetName}',
+    );
+  }
+
+  ReactElement _renderSaveAsMenuItem() {
+    return (SaveAsMenuItem()
+      ..initialValue = _currentStateKeyIsReadOnly ? 'Copy of ${props.currentDataSetName}' : props.currentDataSetName
+      ..onSave = (newStateName) {
+        _handleCurrentLocalStorageStateSaveAs(newStateName: newStateName);
+      }
+    )();
+  }
+
+  ReactElement _renderSavedStateMenuItem(String stateKey) {
+    final json = localTodoAppStorage.mapOfCustomStatesJson[stateKey];
+    return (SavedDataMenuItem()
+      ..key = json['name']
+      ..['selected'] = json['name'] == props.currentDataSetName
+      ..localStorageKey = json['name']
+      ..onSelect = _loadFromLocalStorage
+      ..onRename = (newStateName) {
+        _handleLocalStorageStateRename(newStateName, renamedFrom: json['name']);
+      }
+      ..onDelete = (stateName) {
+        localTodoAppStorage.remove(stateName);
+        forceUpdate();
+      }
+    )();
+  }
+
+  List<ReactElement> _renderSavedStateMenuItems() {
+    if (localTodoAppStorage.mapOfCustomStatesJson.isEmpty) return null;
+
+    return [
+      Divider({'key': 'divider'}),
+      (Dom.li()..key = 'dividerLabel')(
+        Typography({
+          'color': 'textSecondary',
+          'display': 'block',
+          'variant': 'caption',
+          'style': {'margin': '5px 0 0 16px'},
+        },
+          'Custom Data Sets',
+        ),
+      ),
+      ...localTodoAppStorage.mapOfCustomStatesJson.keys.map(_renderSavedStateMenuItem).toList(),
```

Comment:
#nit `.toList` shouldn't be necessary here
```suggestion
      ...localTodoAppStorage.mapOfCustomStatesJson.keys.map(_renderSavedStateMenuItem),
```